### ISSUE TITLE
Event when Plug In enters stationary Mode

### ISSUE THREAD

**Issue by Stophface** [transistorsoft/react-native-background-geolocation#1844]
## Your Environment
Plugin version: ^4.12.0
Platform: iOS or Android iOS
OS version: 16.4
React Native version (react-native -v): 0.71.7
* Plugin config
```javascript <!-- Syntax-highlighting:  paste your code below -->
export const BACKGROUND_GEOLOCATION_READY_CONFIG = {
  desiredAccuracy: BackgroundGeolocation.DESIRED_ACCURACY_MEDIUM,
  stationaryRadius: 25,
  disableElasticity: true,
  distanceFilter: 50,
  disableLocationAuthorizationAlert: true,
  maxRecordsToPersist: 1,
  stopOnTerminate: true,
  stopTimeout: 30,
  disableMotionActivityUpdates: true,
  debug: false,
  logLevel: BackgroundGeolocation.LOG_LEVEL_OFF,
  showsBackgroundLocationIndicator: true,
  allowIdenticalLocations: false,
  foregroundService: true,
};
```

Is there a event listener/subscription or any other method that can notify me, when the Plug In enters the [stationary state](https://github.com/transistorsoft/react-native-background-geolocation/wiki/Philosophy-of-Operation#ios-stationary-state) on iOS (stationary as described in the philosophy of operation). There is `onMotionChange()`, but I guess the `STATIONARY` described in the [docs](https://transistorsoft.github.io/react-native-background-geolocation/classes/backgroundgeolocation.html#onmotionchange) is a different `stationary` as described in the philosophy of operation?

**Comment by christocracy** [comment]
is *is* `onMotionChange`

**Comment by Stophface** [comment]
@christocracy I am not sure if I understand your answere correct. 
When `onMotionChange`  is `!event.isMoving === true`, then its not stopping for a short moment, i.e. at red traffic lights, but the plug in settled into `stationary` mode?

**Comment by christocracy** [comment]
Then have a look at onActivityChange?  In fact, why not read about all the events?  There's only 13.

**Comment by christocracy** [comment]
Tell me what you misunderstand about the docs.

**Comment by Stophface** [comment]
Generally the docs are very well written, expecially in conjuction with with Philosophy of operation part! And your support and the Plug In is great. Thanks a lot!

I truly missunderstood for example `stopOnTerminate`. And my missunderstanding surely wouldn't have been so great, if I would have understood how iOS works under the hood. I have read the documentation on [stopOnTerminate](https://transistorsoft.github.io/react-native-background-geolocation/interfaces/config.html#stoponterminate) in a way that this flag is only used for when the *user* terminates the App and *not* the OS, because it says

> When the **user** terminates the app, the plugin will [BackgroundGeolocation.stop](https://transistorsoft.github.io/react-native-background-geolocation/classes/backgroundgeolocation.html#stop) tracking. Set this to false to continue tracking after application terminate.

However, from your support on GitHub (thanks again!) I learned that the OS *will* terminate the App when its idle in the background for a while. And to me it is not clear, that `stopOnTerminate` also handles this case. So it is not only when user terminates the App, but also the OS. And I always wondered why the tracking did not resume, after putting the App into the background (not terminating it) and setting `stopOnTerminate: true`. That took some solid time to figure out :)


I was also unsure about [onMotionChange](https://transistorsoft.github.io/react-native-background-geolocation/classes/backgroundgeolocation.html#onmotionchange) because it feedbacks either `moving` or `stationary`. From reading the Philosophy of operation I concluded that the `stationary` emitted in `onMotionChange` should be the `stationary`  the Philosphy of operation talks about. But then I looked at the other parameter emitted, which is `moving`... That made me precautios, because the plug in is also active for the duration I define in `stopTimeout`, no matter if I am moving or staying still (the blue bar on iOS is there, notifying me that an app is actively using my location). So I though there should be a third state, `not-moving` or something. And then I got insecure if I concluded what `onMotionChange` is monitoring until you answered me. And I think clearer would be `non-stationary`, `stationary` or something, because `moving` can also be `not-moving` which is not `stationary` neither. 

I hope I expressed myself clear.
